### PR TITLE
Refactor TransactionController unit tests

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## Explanation

Refactor the `TransactionController` unit tests for general simplicity and maintainability, specifically:

- Group tests using describes for each public method.
- Rename tests to remove implied `should` and use updated standard.
- Remove Sinon in favour of Jest mocking.
- Use `it.each` and template functions to avoid duplication in very similar tests.
- Create a single `ACCOUNT_MOCK` constant to avoid duplicate address variables.
- Create a `newController` function to avoid duplicate controller options and config.
- Remove all assertions from within event listeners.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -1292,6 +1136,194 @@ export class TransactionController extends BaseController<
...
+      const rawTransaction = bufferToHex(signedTx.serialize());
+
+      transactionMeta.rawTransaction = rawTransaction;
+      this.updateTransaction(transactionMeta);
+      const transactionHash = await query(this.ethQuery, 'sendRawTransaction', [
+        rawTransaction,
+      ]);
+      transactionMeta.transactionHash = transactionHash;
+      transactionMeta.status = TransactionStatus.submitted;
+      console.log('New Status', TransactionStatus.submitted);
```
Comment (OGPoyraz) [inline packages/transaction-controller/src/TransactionController.ts:1292]: Is this a forgotten one or intentional?

Comment (matthewwalsh0) [reply packages/transaction-controller/src/TransactionController.ts]: Forgotten, and removed by the merge.

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import * as sinon from 'sinon';
```
Comment (vinistevam) [inline packages/transaction-controller/src/TransactionController.test.ts:1]: :heart:

**Review by vinistevam — APPROVED** [review]

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 3]
```diff
@@ -384,1453 +386,1123 @@ const ETH_TX_HISTORY_DATA_GOERLI_NO_TRANSACTIONS_FOUND = {
...
-        },
-        {
-          interval: 5000,
-          sign: async (transaction: any) => transaction,
-          txHistoryLimit: 1,
+    it('verifies transactions using the correct blockchain', async () => {
+      const controller = newController();
+
+      controller.state.transactions.push({
+        from: MOCK_PRFERENCES.state.selectedAddress,
```
Comment (vinistevam) [inline packages/transaction-controller/src/TransactionController.test.ts:1189]: small typo, probably it was already here before the refactor.

**Pull request merged**